pkg/operator/storageclasscontroller/storageclasscontroller.go

```diff
@@ -0,0 +1,215 @@
+package storageclasscontroller
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"gopkg.in/gcfg.v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/client-go/kubernetes"
+	corelister "k8s.io/client-go/listers/core/v1"
+	"k8s.io/klog/v2"
+	"k8s.io/legacy-cloud-providers/vsphere"
+
+	v1 "github.com/openshift/api/operator/v1"
+	configinformers "github.com/openshift/client-go/config/informers/externalversions"
+	infralister "github.com/openshift/client-go/config/listers/config/v1"
+	"github.com/openshift/library-go/pkg/controller/factory"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceread"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
+
+	"github.com/vmware/govmomi"
+	"github.com/vmware/govmomi/find"
+	"github.com/vmware/govmomi/property"
+	"github.com/vmware/govmomi/vapi/tags"
+	"github.com/vmware/govmomi/vim25/mo"
+)
+
+const (
+	infraGlobalName       = "cluster"
+	cloudConfigNamespace  = "openshift-config"
+	DatastoreInfoProperty = "info"
+	SummaryProperty       = "summary"
+	resyncDuration        = 5 * time.Minute
+)
+
+type StorageClassController struct {
+	name            string
+	targetNamespace string
+	manifest        []byte
+	kubeClient      kubernetes.Interface
+	operatorClient  v1helpers.OperatorClient
+	configMapLister corelister.ConfigMapLister
+	secretLister    corelister.SecretLister
+	infraLister     infralister.InfrastructureLister
+}
+
+type vSphereConnection struct {
+	client     *govmomi.Client
+	config     *vsphere.VSphereConfig
+	tagManager *tags.Manager
+}
+
+func NewStorageClassController(
+	name,
+	targetNamespace string,
+	manifest []byte,
+	kubeClient kubernetes.Interface,
+	kubeInformers v1helpers.KubeInformersForNamespaces,
+	operatorClient v1helpers.OperatorClient,
+	configInformer configinformers.SharedInformerFactory,
+	recorder events.Recorder,
+) factory.Controller {
+	configMapInformer := kubeInformers.InformersFor(cloudConfigNamespace).Core().V1().ConfigMaps()
+	secretInformer := kubeInformers.InformersFor(targetNamespace).Core().V1().Secrets()
+	infraInformer := configInformer.Config().V1().Infrastructures()
+	rc := recorder.WithComponentSuffix("vmware-" + strings.ToLower(name))
+	c := &StorageClassController{
+		name:            name,
+		targetNamespace: targetNamespace,
+		manifest:        manifest,
+		kubeClient:      kubeClient,
+		operatorClient:  operatorClient,
+		configMapLister: configMapInformer.Lister(),
+		secretLister:    secretInformer.Lister(),
+		infraLister:     infraInformer.Lister(),
+	}
+	return factory.New().WithInformers(
+		configMapInformer.Informer(),
+		secretInformer.Informer(),
+		infraInformer.Informer(),
+		operatorClient.Informer(),
```

Comment:
What do you think about adding a StorageClass informer here as well, so that this controllers resyncs when something changes a StorageClass?

Reply:
We could, but we only created namespaced informers in this operator. We could do this in a follow up.

Reply:
ack, I'll add it to my TODO